LESSONS.md

```diff
@@ -0,0 +1,7 @@
+2020-08-18 08:49:56
+
+# Lessons Learned
+
+-   TypeSet vs TypeList
```

Comment:
Honestly, I'm a little hesitant to do that mostly because I still feel like its a matter of having more in-depth knowledge of the underlying behavior. But I guess I wouldn't be opposed to saying something like "if the attribute can be a TypeList, then go with TypeList. We've found attributes that use TypeList to be bit less nuanced and easier to maintain and still accomplish the same goal as TypeSet"... please fact check me on the "still accomplish the same goal" part 😉

Reply:
From the Extending Terraform guide:

TypeSet
> TypeSet implements set behavior and is used to represent an unordered collection of items, meaning that their ordering specified does not need to be consistent, and the ordering itself has no impact on the behavior of the resource.

TypeList:
> Used to represent an ordered collection of items, where the order the items are presented can impact the behavior of the resource being modeled. An example of ordered items would be network routing rules, where rules are examined in the order they are given until a match is found. The items are all of the same type defined by the Elem property.

Reply:
I don't think we should recommend one over the other - they both have their uses.

It probably is worth pointing out that the idiom for nesting schema structures is `TypeList` with `MaxItems: 1` as discussed [here](https://github.com/hashicorp/terraform-plugin-sdk/issues/155#issuecomment-489699737).